packages/typir/src/features/printing.ts

```diff
@@ -28,7 +28,10 @@ export interface ProblemPrinter {
     printTypirProblems(problems: TypirProblem[]): string;
 
     printDomainElement(domainElement: unknown, sentenceBegin: boolean): string;
-    printType(type: Type): string;
+    /** This function should be used by other services, not type.getName(). */
```

Comment:
Why exactly? The function only calls getName, doesn't it? Do you expect changes to that in the future?

Reply:
Yes, the default implementation is very easy. But maybe, a customization of type names might be required in some applications, e.g. the name with some HTML tags. Than it is much easier to do the customization by only overriding this function.

I added more comments.